code/game/machinery/vending.dm

```diff
@@ -2374,6 +2374,6 @@ var/global/num_vending_terminals = 1
 		else
 			to_chat(user, "<span class='notice'>It doesn't fit.</span>")
 		return
-	if (istype(W, /obj/item/weapon/reagent_containers/food/snacks/chococoin)
+	if (istype(W, /obj/item/weapon/reagent_containers/food/snacks/chococoin))
```

Comment:
Instead of just continuing to make and modify typechecks, you should make a proc /is_valid_coin(obj/item/weapon/W)

You can then override in the Jewvend